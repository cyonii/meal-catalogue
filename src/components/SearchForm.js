import { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import { setFilter, setQuery } from '../redux/actions/metaData';

const SearchForm = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const [searchInput, setSearchInput] = useState('');

  const handleQueryInput = (e) => {
    setSearchInput(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(setQuery(searchInput));
    dispatch(setFilter('search'));
    history.push('/');
  };

  return (
    <Form className="d-flex" onSubmit={handleSubmit}>
      <Form.Control
        type="search"
        placeholder="Movie or show title"
        className="rounded-0 border-0"
        aria-label="Movie or show title"
        onChange={handleQueryInput}
      />
      <Button type="submit" variant="success" className="rounded-0 no-outline">Search</Button>
    </Form>
  );
};

export default SearchForm;
