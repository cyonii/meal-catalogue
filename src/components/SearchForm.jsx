import { useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";
import { setFilter, setQuery } from "../redux/actions/metaData";

const SearchForm = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const [searchInput, setSearchInput] = useState("");
  const [disabled, setDisabled] = useState(true);
  const minSearchLength = 3;

  useEffect(() => {
    if (searchInput.length < minSearchLength) {
      setDisabled(true);
    } else {
      setDisabled(false);
    }
  }, [searchInput]);

  const handleQueryInput = (e) => setSearchInput(e.target.value.trimStart());

  const handleSubmit = (e) => {
    e.preventDefault();

    if (searchInput.length < minSearchLength) return;

    dispatch(setQuery(searchInput));
    dispatch(setFilter("search"));
    history.push("/");
  };

  return (
    <form className="flex" onSubmit={handleSubmit} data-testid="search-form">
      <input
        type="search"
        className="dark:bg-gray-700 dark:text-slate-100 focus:bg-gray-500"
        placeholder="Movie or show title"
        aria-label="Movie or show title"
        value={searchInput}
        onInput={handleQueryInput}
      />
      <button
        type="submit"
        disabled={disabled}
        className="btn-green-600 rounded-none"
      >
        Search
      </button>
    </form>
  );
};

export default SearchForm;
