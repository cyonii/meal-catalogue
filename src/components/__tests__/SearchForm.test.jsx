import { render, screen, within } from "@testing-library/react";
import SearchForm from "../SearchForm";

vi.mock("react-redux", () => ({
  useDispatch: () => vi.fn(),
}));

describe("SearchForm", () => {
  it("renders correctly", () => {
    render(<SearchForm />);

    const searchForm = screen.getByTestId("search-form");

    expect(searchForm).toBeInTheDocument();
    expect(searchForm).toMatchSnapshot();
    expect(
      within(searchForm).getByPlaceholderText("Movie or show title").tagName,
    ).toBe("INPUT");
    expect(within(searchForm).getByText("Search").tagName).toBe("BUTTON");
  });
});
