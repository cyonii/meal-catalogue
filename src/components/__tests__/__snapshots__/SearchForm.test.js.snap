// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<form
  class="d-flex"
  data-testid="search-form"
>
  <input
    aria-label="Movie or show title"
    class="rounded-0 border-0 form-control"
    placeholder="Movie or show title"
    type="search"
  />
  <button
    class="rounded-0 no-outline btn btn-success"
    type="submit"
  >
    Search
  </button>
</form>
`;
